src/commands.rs

```diff
@@ -4826,12 +4814,50 @@ fn cmd_git_push(
     let git_repo = get_git_repo(repo.store())?;
     git::push_updates(&git_repo, &args.remote, &ref_updates)
         .map_err(|err| CommandError::UserError(err.to_string()))?;
-    let mut tx = workspace_command.start_transaction("import git refs");
     git::import_refs(tx.mut_repo(), &git_repo)?;
     workspace_command.finish_transaction(ui, tx)?;
     Ok(())
 }
 
+fn branch_updates_for_push(
+    repo: RepoRef,
+    remote_name: &str,
+    branch_name: &str,
+) -> Result<Option<BranchPushUpdate>, CommandError> {
+    let maybe_branch_target = repo.view().get_branch(branch_name);
+    if maybe_branch_target.is_none() {
+        return Err(CommandError::UserError(format!(
+            "Branch {} doesn't exist",
+            branch_name
+        )));
+    }
+    let branch_target = maybe_branch_target.unwrap();
```

Comment:
```suggestion
    let branch_target = match maybe_branch_target {
        Some(branch_target) => branch_target,
        None => {
            return Err(CommandError::UserError(format!(
                "Branch {} doesn't exist",
                branch_name
            )));
        }
    };
```

Reply:
Ah, good point that this could be simpler and avoid the use of `unwrap()`. I went with `ok_or_else()` instead of the `match`. Let me know if you prefer the match.